backend/src/main/java/com/wpz/rbs/service/ReservationService.java

```diff
@@ -0,0 +1,65 @@
+package com.wpz.rbs.service;
+
+import com.wpz.rbs.model.Reservation;
+import com.wpz.rbs.model.reservation.CreateReservation;
+import com.wpz.rbs.repository.ActivityRepository;
+import com.wpz.rbs.repository.ReservationRepository;
+import com.wpz.rbs.repository.RoomRepository;
+import com.wpz.rbs.utils.StaticHelpers;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+
+import java.text.ParseException;
+import java.util.Date;
+import java.util.List;
+
+@Service
+public class ReservationService {
+    private final ReservationRepository reservationRepository;
+    private final ActivityRepository activityRepository;
+    private final RoomRepository roomRepository;
+
+    public ReservationService(ReservationRepository reservationRepository, ActivityRepository activityRepository, RoomRepository roomRepository) {
+        this.reservationRepository = reservationRepository;
+        this.activityRepository = activityRepository;
+        this.roomRepository = roomRepository;
+    }
+
+    public Reservation getById(int id) {
+        return reservationRepository.findById(id).get();
+    }
+
+    public List<Reservation> getByRoomId(int roomId) {
+        return reservationRepository.findAllByRoom_Id(roomId);
+    }
+
+    public List<Reservation> getAll() {
+        return reservationRepository.findAll();
+    }
+
+    public ResponseEntity<?> createReservation(CreateReservation reservation) throws ParseException {
+        Date startDate = StaticHelpers.parseDateTime(reservation.getStart_time());
+        Date endDate = StaticHelpers.parseDateTime(reservation.getEnd_time());
+        if (!roomRepository.existsById(reservation.getRoom_id()))
+            return ResponseEntity.status(400).body("Wrong room id");
+        if (startDate.compareTo(endDate) >= 0)
+            return ResponseEntity.status(409).body("Start date is later than end date");
+        if (collisionExists(reservation.getRoom_id(), startDate, endDate))
+            throw new RuntimeException("Room reservation collision detected");
+        return ResponseEntity.ok(reservationRepository.save(new Reservation(reservation)).getId());
+    }
+
+    private boolean collisionExists(int roomId, Date newStartDate, Date newEndDate) {
```

Comment:
Can we do this in sql on database. Will be faster than returning all activities.